core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -229,4 +229,11 @@ private[spark] object UI {
     .stringConf
     .transform(_.toUpperCase(Locale.ROOT))
     .createWithDefault("LOCAL")
+
+  val UI_SQL_GROUP_SUB_EXECUTION_ENABLED = ConfigBuilder("spark.ui.sql.group.sub.execution.enabled")
```

Comment:
This PR introduces 4 config namespace groups like the following. Shall we simplify the config namespace?
```
spark.ui.sql.*
spark.ui.sql.group.*
spark.ui.sql.group.sub.*
spark.ui.sql.group.sub.execution.*
```

Reply:
changed to `spark.ui.sql.groupSubExecutionEnabled` but I'm glad to take any other naming suggestions. :)